File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    const answer = []
    const day = []
    let cnt = 1
    
    // 작업 일수
    progresses.forEach((data, index) => {
        let count = 0
        while(data < 100) {
            data += speeds[index]
            count++
        }
        day.push(count)
    })
    
    // 현재
    let max = day[0]
    
    for(let i = 1; i<day.length; i++) {
        
        if(max >= day[i]) {
            cnt++
        } else {
            max = day[i]
            answer.push(cnt)
            cnt = 1
        }
    }
    
    answer.push(cnt)
     
    return answer

}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
from collections import deque
def solution(progresses, speeds):
    
    answer = []
    qu = deque()
    
    
    for task, speed in zip(progresses, speeds):
        task_count = 0
        
        while task < 100:
            task += speed
            task_count += 1
            
        qu.append(task_count)
    
    while qu:
        count = 1
        day = qu.popleft()
        
        while qu and day >= qu[0]:
            qu.popleft()
            count += 1
        answer.append(count)
    
    return answer
        
        
        